File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        numList = {}
        for num in nums:
            if num not in numList:
                numList[num] = 1
            else:
                numList[num] += 1
        for val in numList:
            if numList[val] > len(nums) // 2:
                return val

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> seen;
        for (const auto& x: nums)
        {
            if (!seen.contains(x))
            {
                seen.insert(x);
            }
            else {
                return true;
            }
        }
        return false;
        
    }
};